manticore/core/smtlib/solver.py

```diff
@@ -351,15 +344,30 @@ def _assert(self, expression: Bool):
 
     def __getvalue_bv(self, expression_str: str) -> int:
         self._smtlib.send(f"(get-value ({expression_str}))")
-        pattern, base = self._get_value_fmt
-        m = pattern.match(self._smtlib.recv())
+        t = self._smtlib.recv()
+        pattern = RE_GET_EXPR_VALUE_FMT_BIN
+        base = 2
+        m = pattern.match(t)
+        if m is None:
+            pattern = RE_GET_EXPR_VALUE_FMT_DEC
+            m = pattern.match(t)
+            base = 10
+        if m is None:
+            pattern = RE_GET_EXPR_VALUE_FMT_HEX
+            m = pattern.match(t)
+            base = 16
```

Comment:
Minor nit, we could change to the following:
```suggestion
        base = 2
        m = RE_GET_EXPR_VALUE_FMT_BIN.match(t)
        if m is None:
            m = RE_GET_EXPR_VALUE_FMT_DEC.match(t)
            base = 10
        if m is None:
            m = RE_GET_EXPR_VALUE_FMT_HEX.match(t)
            base = 16
```